### PR TITLE
Task/grow 551 connect channel upgrade modal

### PR DESCRIPTION
**Pull request by daisymarie128**
JIRA ticket: [GROW-551](https://buffer.atlassian.net/browse/GROW-551)

Screen Shot 2022-06-06 at 2 26 40 pm

### CODE REVIEW

File: packages/app-shell/src/common/graphql/billing.js [hunk 1]
```diff
@@ -79,6 +79,27 @@ export const UPDATE_SUBSCRIPTION_QUANTITY = gql`
   }
 `;
 
+export const UPDATE_SUBSCRIPTION_QUANTITY_BY_CHANNEL_IDS = gql`
+  mutation updateQuantityByChannelIds(
+    $organizationId: String
+    $channelIds: [String]
```
Comment (daisymarie128) [inline packages/app-shell/src/common/graphql/billing.js:85]: 🏁 Check this syntax is correct

File: packages/app/src/App.js [hunk 2]
```diff
@@ -112,6 +112,22 @@ const ModalTesting = () => {
...
         Render Start Trial Button
       </button>
+
+      <h3>Render Connect Channel Upgrade</h3>
+      <button
+        id="render_connect_channel_upgrade"
+        onClick={() => {
+          const { MODALS, actions } = window?.appshell || {};
+          actions.openModal(MODALS.connectChannelUpgrade, {
+            cta: 'connect_channel_upgrade',
```
Comment (dinostheo) [inline packages/app/src/App.js:122]: Not sure if this CTA is proper.

File: packages/app-shell/src/components/Modal/modals/ConnectChannelUpgrade/components/CardBody.jsx [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+  planId,
+  planCycle,
+  user,
+  onSuccess,
+  openPlanSelector,
+  closeModal,
+  cta,
+  trialInfo,
+}) => {
+  const newChannelQuantity = quantity + 1;
```
[the remark this answers is not in the document]
Comment (daisymarie128) [reply packages/app-shell/src/components/Modal/modals/ConnectChannelUpgrade/components/CardBody.jsx]: Ah yep! Good catch @dinostheo that logic was there when I first made a demo of the modal, il remove this and count the number of IDs which get passed through 🙌

File: packages/app-shell/src/components/Modal/modals/ConnectChannelUpgrade/components/CardBody.jsx [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+
+  const {
+    useUpdateSubscriptionQuantityByChannelIds,
+    data: useUpdateSubscriptionQuantityData,
+    isProcessing,
+    error,
+  } = useUpdateSubscriptionQuantityByChannelIds({
+    user,
+    channelIds: [], // TODO: Need to pass in array of ids here
```
[the remark this answers is not in the document]
Comment (daisymarie128) [reply packages/app-shell/src/components/Modal/modals/ConnectChannelUpgrade/components/CardBody.jsx]: Ah! Good point @dinostheo il write a new util function which can calculate this 👍

File: packages/app/src/App.js (hunk 2 above)
Comment (daisymarie128) [reply packages/app/src/App.js:130]: Ooh ok, I think I've forgotten the process for creating new ctas @dinostheo . Can you remind me what steps are needed to add a new one? 🤔

File: packages/app-shell/src/components/Modal/modals/ConnectChannelUpgrade/components/CardBody.jsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (daisymarie128) [reply packages/app-shell/src/components/Modal/modals/ConnectChannelUpgrade/components/CardBody.jsx]: @dinostheo curiously is this also being handled on the BE side when we make the request? 🤔 From what I could see in the modal, we only use our calculation to validate the confirm buttons state and we only pass back the list of channel ids to the `useUpdateSubscriptionQuantityByChannelIds` request 🧐 does that look right to you? 

cc @cheahenry

File: packages/app/src/App.js (hunk 2 above)
Comment (cheahenry) [reply packages/app/src/App.js:130]: We need to follow this [convention](https://www.notion.so/buffer/CTA-Parameter-Conventions-Call-to-Action-0ae87bdd99574e3888afb2fe6b75035a) for CTAs

**Pull request merged**